src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,91 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import { TodoStatus } from '../../types/TodoStatus';
+
+interface Props {
+  searchQuery: string,
+  setSearchQuery: React.Dispatch<React.SetStateAction<string>>
+  todoStatus: TodoStatus,
+  setTodoStatus: React.Dispatch<React.SetStateAction<TodoStatus>>,
+}
+
+export const TodoFilter: React.FC<Props> = ({
+  searchQuery,
+  setSearchQuery,
+  todoStatus,
+  setTodoStatus,
+}) => {
+  const handleClick = () => {
+    setSearchQuery('');
+  };
+
+  const handleSelectChange = (
+    event :React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    switch (event.target.value) {
+      case 'all':
+        setTodoStatus(TodoStatus.ALL);
+        break;
+      case 'completed':
+        setTodoStatus(TodoStatus.COMPLETED);
+        break;
+      case 'active':
+        setTodoStatus(TodoStatus.ACTIVE);
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  const handleInputChange = (
```

Comment:
in handleClick we have mouse event (MouseEvent<HTMLButtonElement, MouseEvent>) in input we have React.ChangeEvent<HTMLInputElement> and when we create 1 handler for input and button, this will call errors with ts